src/encoded/static/components/search.js

```diff
@@ -735,6 +735,289 @@ TypeTerm.propTypes = {
+        this.toggleDateFacet = this.toggleDateFacet.bind(this);
+        this.setActiveFacetParameters = this.setActiveFacetParameters.bind(this);
+        this.handleReset = this.handleReset.bind(this);
+        this.resetMonthDropDowns = this.resetMonthDropDowns.bind(this);
+    }
+
+    componentDidMount() {
+        this.setActiveFacetParameters();
+    }
+
+    setActiveFacetParameters() {
+        // Filter 'facets' to get selected 'facet' object (either 'date_released' or 'date_submitted' object)
+        const activeFacet = this.props.facets.filter(f => f.field === this.state.activeFacet)[0];
+        // Sorted terms for the selected facet object
+        const activeFacetTerms = _.sortBy(activeFacet.terms, obj => moment(obj.key, 'YYYY-MM-DD').toISOString());
+
+        // All possible years with results
+        const possibleYears = activeFacetTerms.map(term => term.key.split('-')[0]).filter((item, i, ar) => (ar.indexOf(item) === i));
+        this.setState({ possibleYears });
+
+        // Set dropdown lists to be full lists of possiblities
+        this.setState({ startYears: possibleYears });
+        this.setState({ endYears: possibleYears });
+        this.setState({ startMonths: this.state.possibleMonths });
+        this.setState({ endMonths: this.state.possibleMonths });
+
+        // Set dropdowns to be initialized to boundaries of full range
+        this.setState({ startYear: possibleYears[0] });
+        this.setState({ endYear: possibleYears[possibleYears.length - 1] });
+        this.setState({ startMonth: '01' });
+        this.setState({ endMonth: '12' });
+    }
+
+    selectYear(event) {
+        // We are changing the start year, which means we need to change the possibilities for the end years and also the possible start months
+        if (event.target.id === 'select-start-year') {
+            // Set startYear to be user choice
+            this.setState({ startYear: event.target.value }, () => {
+                // Check if now the years match and month lists need to be limited
+                this.checkForSameYear();
+            });
+            // Possibilities for endYears must now all be greater than the new startYear
+            const endYears = this.state.possibleYears.filter(year => +year >= event.target.value);
+            this.setState({ endYears });
+        // We are changing the end year, which means we need to change the possiblities for the starting year and also the possible end months
+        } else {
+            // Set endYear to be user choice
+            this.setState({ endYear: event.target.value }, () => {
+                // Check if now the years match and month lists need to be limited
+                this.checkForSameYear();
+            });
+            // Possiblities for startYears must now all be less than the new endYears
+            const startYears = this.state.possibleYears.filter(year => +year <= event.target.value);
```

Comment:
Have to say `=> +year <=` is the coolest syntax i’ve seen. event.target.value can be a number? I always thought they were strings no matter what, but I think this wouldn’t work at all if that were true.

Reply:
Thanks! It is actually a little weird that this works because `event.target.value` is actually a string. Probably I should just convert it for clarity but the reason it works is that +year is a number and from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comparison_Operators): "When comparing a number and a string, the string is converted to a number value."